### DISCUSSION TITLE
Why is TransactionBuilder::new private?

### DISCUSSION THREAD

**Discussion by lsunsi** [weiznich/diesel_async#145] [Q&A] [answered]
I'm trying to implement a wrapper for the Connection locally in order to implement some telemetry while I wait for the release of the new API. I'm basing my implementation off of diesel-tracing, which uses `TransactionBuilder::new` on the sync diesel library to provide the `fn build_transaction` implementation.

My question is: is there an specific reason this `new` function is not (1) public or (2) gated like in the sync counterpart? Being public might make sense since diesel_async itself is considered pre-release, but being gated could make sense as well and enable the kind of implementation I'm trying to achieve.

Thanks and sorry if it's a dumb question

**Comment by weiznich** [comment, marked as answer]
That function is private as the corresponding sync equivalent not part of the public API of diesel itself and `diesel_async` does not have the equivalent of the unstable feature flag. 

That written I might be open to expose it if someone submits a PR that changes it to pub, adds documentation and covers that function with a relevant test.

**Reply by lsunsi** [reply]
This is awesome, I might take a crack at it. Thanks!
